### PR TITLE
Check aspect mask in vkCmdClearAttachments

### PR DESCRIPTION
**Pull request by chrisforbes**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -8527,6 +8533,23 @@ VKAPI_ATTR void VKAPI_CALL CmdClearAttachments(VkCommandBuffer commandBuffer, ui
...
+
+            if (image_view) {
+                auto image_view_state = getImageViewState(dev_data, image_view);
+                auto aspects_present = image_view_state->create_info.subresourceRange.aspectMask;
+                auto extra_aspects = clear_desc->aspectMask & ~aspects_present;
+
+                if (extra_aspects) {
+                    skip_call |= log_msg(
+                            dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_IMAGE_VIEW_EXT,
+                            reinterpret_cast<uint64_t &>(image_view), __LINE__, DRAWSTATE_INVALID_IMAGE_ASPECT, "DS",
```
Comment (tobine) [inline layers/core_validation.cpp]: Should use VALIDATION_ERROR_01125 I believe

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -8499,25 +8499,31 @@ VKAPI_ATTR void VKAPI_CALL CmdClearAttachments(VkCommandBuffer commandBuffer, ui
...
                         (uint64_t)commandBuffer, __LINE__, DRAWSTATE_MISSING_ATTACHMENT_REFERENCE, "DS",
                         "vkCmdClearAttachments() color attachment index %d out of range for active subpass %d; ignored",
-                        attachment->colorAttachment, pCB->activeSubpass);
+                        clear_desc->colorAttachment, pCB->activeSubpass);
                 }
-                else if (pSD->pColorAttachments[attachment->colorAttachment].attachment == VK_ATTACHMENT_UNUSED) {
+                else if (pSD->pColorAttachments[clear_desc->colorAttachment].attachment == VK_ATTACHMENT_UNUSED) {
                     skip_call |= log_msg(
                         dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
                         (uint64_t)commandBuffer, __LINE__, DRAWSTATE_MISSING_ATTACHMENT_REFERENCE, "DS",
```
Comment (tobine) [inline layers/core_validation.cpp]: Should also use VALIDATION_ERROR_01114 I believe

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -8499,25 +8499,31 @@ VKAPI_ATTR void VKAPI_CALL CmdClearAttachments(VkCommandBuffer commandBuffer, ui
...
 
-        for (uint32_t attachment_idx = 0; attachment_idx < attachmentCount; attachment_idx++) {
-            const VkClearAttachment *attachment = &pAttachments[attachment_idx];
-            if (attachment->aspectMask & VK_IMAGE_ASPECT_COLOR_BIT) {
-                if (attachment->colorAttachment >= pSD->colorAttachmentCount) {
+            if (clear_desc->aspectMask & VK_IMAGE_ASPECT_COLOR_BIT) {
+                if (clear_desc->colorAttachment >= pSD->colorAttachmentCount) {
                     skip_call |= log_msg(
                         dev_data->report_data, VK_DEBUG_REPORT_ERROR_BIT_EXT, VK_DEBUG_REPORT_OBJECT_TYPE_COMMAND_BUFFER_EXT,
                         (uint64_t)commandBuffer, __LINE__, DRAWSTATE_MISSING_ATTACHMENT_REFERENCE, "DS",
```
Comment (tobine) [inline layers/core_validation.cpp]: Should use VALIDATION_ERROR_01114 I believe

File: layers/core_validation.cpp (hunk 1 above)
Comment (karl-lunarg) [inline layers/core_validation.cpp]: And don't forget to update vk_validation_error_database.txt.

File: layers/core_validation.cpp (hunk 2 above)
Comment (chrisforbes) [inline layers/core_validation.cpp]: This is one of the cases we were going to relax -- it's not really an error at all.

**Comment by chrisforbes** [comment]
Updated

File: layers/vk_validation_error_database.txt [hunk 4]
```diff
@@ -1104,7 +1104,7 @@ VALIDATION_ERROR_01110~^~U~^~Unknown~^~vkCmdClearDepthStencilImage~^~For more in
...
-VALIDATION_ERROR_01114~^~U~^~Unknown~^~vkCmdClearAttachments~^~For more information refer to Vulkan Spec Section '17.2. Clearing Images Inside A Render Pass Instance' which states 'If the aspectMask member of any given element of pAttachments contains VK_IMAGE_ASPECT_COLOR_BIT, the colorAttachment member of those elements must refer to a valid color attachment in the current subpass' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#vkCmdClearAttachments)~^~
+VALIDATION_ERROR_01114~^~Y~^~Unknown~^~vkCmdClearAttachments~^~For more information refer to Vulkan Spec Section '17.2. Clearing Images Inside A Render Pass Instance' which states 'If the aspectMask member of any given element of pAttachments contains VK_IMAGE_ASPECT_COLOR_BIT, the colorAttachment member of those elements must refer to a valid color attachment in the current subpass' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#vkCmdClearAttachments)~^~
```
Comment (tobine) [inline layers/vk_validation_error_database.txt]: I believe the test for this case is "MissingClearAttachment"

File: layers/vk_validation_error_database.txt [hunk 5]
```diff
@@ -1115,7 +1115,7 @@ VALIDATION_ERROR_01121~^~U~^~Unknown~^~vkCmdClearAttachments~^~For more informat
...
 VALIDATION_ERROR_01124~^~U~^~Unknown~^~vkCmdClearAttachments~^~For more information refer to Vulkan Spec Section '17.2. Clearing Images Inside A Render Pass Instance' which states 'rectCount must be greater than 0' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#vkCmdClearAttachments)~^~
-VALIDATION_ERROR_01125~^~U~^~Unknown~^~vkCmdClearAttachments~^~For more information refer to Vulkan Spec Section '17.2. Clearing Images Inside A Render Pass Instance' which states 'If aspectMask includes VK_IMAGE_ASPECT_COLOR_BIT, it must not include VK_IMAGE_ASPECT_DEPTH_BIT or VK_IMAGE_ASPECT_STENCIL_BIT' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#VkClearAttachment)~^~
+VALIDATION_ERROR_01125~^~Y~^~Unknown~^~vkCmdClearAttachments~^~For more information refer to Vulkan Spec Section '17.2. Clearing Images Inside A Render Pass Instance' which states 'If aspectMask includes VK_IMAGE_ASPECT_COLOR_BIT, it must not include VK_IMAGE_ASPECT_DEPTH_BIT or VK_IMAGE_ASPECT_STENCIL_BIT' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/xhtml/vkspec.html#VkClearAttachment)~^~
```
Comment (tobine) [inline layers/vk_validation_error_database.txt]: I don't believe there is a test for this case so test column should be "None."

Comment (chrisforbes) [inline layers/vk_validation_error_database.txt]: OK, I will do this, and add a test that covers it in a followup PR.

**Pull request merged**